### PR TITLE
Kconfig dt type

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: doc/guides/kconfig/index.rst [hunk 1]
```diff
@@ -820,17 +820,21 @@ details on the functions.
...
-  dt_node_reg_addr(kconf, _, path, index=0, unit=None):
-  dt_node_reg_size(kconf, _, path, index=0, unit=None):
+  dt_chosen_reg_addr_int(kconf, _, chosen, index=0, unit=None):
+  dt_chosen_reg_size_int(kconf, _, chosen, index=0, unit=None):
+  dt_chosen_reg_addr_hex(kconf, _, chosen, index=0, unit=None):
+  dt_chosen_reg_size_hex(kconf, _, chosen, index=0, unit=None):
+  dt_node_reg_addr_int(kconf, _, path, index=0, unit=None):
+  dt_node_reg_size_int(kconf, _, path, index=0, unit=None):
+  dt_node_reg_addr_hex(kconf, _, path, index=0, unit=None):
+  dt_node_reg_size_hex(kconf, _, path, index=0, unit=None):
```
Comment (ulfalizer) [inline doc/guides/kconfig/index.rst]: Maybe slightly easier to read in this order:

```
dt_chosen_reg_addr_int(kconf, _, chosen, index=0, unit=None):
dt_chosen_reg_addr_hex(kconf, _, chosen, index=0, unit=None):
dt_chosen_reg_size_int(kconf, _, chosen, index=0, unit=None):
dt_chosen_reg_size_hex(kconf, _, chosen, index=0, unit=None):
dt_node_reg_addr_int(kconf, _, path, index=0, unit=None):
dt_node_reg_addr_hex(kconf, _, path, index=0, unit=None):
dt_node_reg_size_int(kconf, _, path, index=0, unit=None):
dt_node_reg_size_hex(kconf, _, path, index=0, unit=None):
```

File: scripts/kconfig/kconfigfunctions.py [hunk 2]
```diff
@@ -288,8 +318,7 @@ def dt_compat_enabled(kconf, _, compat):
         return "n"
 
     for node in edt.nodes:
-        if compat in node.compats and node.enabled:
-            return "y"
+        if compat in node.compats and node.enabled: return "y"
```
Comment (ulfalizer) [inline scripts/kconfig/kconfigfunctions.py]: Accidental change?

File: scripts/kconfig/kconfigfunctions.py [hunk 3]
```diff
@@ -300,9 +329,13 @@ def dt_compat_enabled(kconf, _, compat):
...
-        "dt_node_reg_addr": (dt_node_reg_addr, 1, 3),
-        "dt_node_reg_size": (dt_node_reg_size, 1, 3),
+        "dt_chosen_reg_addr_int": (dt_chosen_reg, 1, 4),
+        "dt_chosen_reg_size_int": (dt_chosen_reg, 1, 4),
+        "dt_chosen_reg_addr_hex": (dt_chosen_reg, 1, 4),
+        "dt_chosen_reg_size_hex": (dt_chosen_reg, 1, 4),
+        "dt_node_reg_addr_int": (dt_node_reg, 1, 4),
+        "dt_node_reg_size_int": (dt_node_reg, 1, 4),
+        "dt_node_reg_addr_hex": (dt_node_reg, 1, 4),
+        "dt_node_reg_size_hex": (dt_node_reg, 1, 4),
```
Comment (ulfalizer) [inline scripts/kconfig/kconfigfunctions.py]: Maybe slightly easier to read in this order:

```python
"dt_chosen_reg_addr_int": (dt_chosen_reg, 1, 4),
"dt_chosen_reg_addr_hex": (dt_chosen_reg, 1, 4),
"dt_chosen_reg_size_int": (dt_chosen_reg, 1, 4),
"dt_chosen_reg_size_hex": (dt_chosen_reg, 1, 4),
"dt_node_reg_addr_int": (dt_node_reg, 1, 4),
"dt_node_reg_addr_hex": (dt_node_reg, 1, 4),
"dt_node_reg_size_int": (dt_node_reg, 1, 4),
"dt_node_reg_size_hex": (dt_node_reg, 1, 4),
```

File: doc/guides/kconfig/index.rst [hunk 4]
```diff
@@ -820,17 +820,21 @@ details on the functions.
```
Comment (ulfalizer) [inline doc/guides/kconfig/index.rst]: Could maybe add something like

```
The `*_int` version of each function return the value as a decimal integer, while the `*_hex` version returns a hexidecimal value starting with "0x".
```

**Review by ulfalizer — CHANGES_REQUESTED** [review]
Looks good to me. Just some nits.

File: boards/arm/actinius_icarus/Kconfig.defconfig [hunk 5]
```diff
@@ -53,17 +53,17 @@ DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
...
 
 if BOARD_ACTINIUS_ICARUS_NS
 
 config FLASH_LOAD_OFFSET
-	default $(dt_chosen_reg_addr,$(DT_CHOSEN_Z_CODE_PARTITION))
+	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
 
 config FLASH_LOAD_SIZE
-	default $(dt_chosen_reg_size,$(DT_CHOSEN_Z_CODE_PARTITION))
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
```
Comment (ulfalizer) [inline boards/arm/actinius_icarus/Kconfig.defconfig]: Think this was supposed to be `dt_chosen_reg_addr_int`.

File: doc/guides/kconfig/index.rst (hunk 1 above)
Comment (galak) [reply doc/guides/kconfig/index.rst]: done

File: scripts/kconfig/kconfigfunctions.py (hunk 2 above)
Comment (galak) [reply scripts/kconfig/kconfigfunctions.py]: yep

File: scripts/kconfig/kconfigfunctions.py (hunk 3 above)
Comment (galak) [reply scripts/kconfig/kconfigfunctions.py]: done

File: doc/guides/kconfig/index.rst (hunk 4 above)
Comment (galak) [reply doc/guides/kconfig/index.rst]: added

**Review by dbkinder — APPROVED** [review]
+1 for docs

File: doc/guides/kconfig/index.rst [hunk 6]
```diff
@@ -816,21 +816,27 @@ Devicetree Related Functions
 ============================
 
 See the Python docstrings in ``scripts/kconfig/kconfigfunctions.py`` for more
-details on the functions.
+details on the functions.  The `*_int` version of each function return the
```
Comment (ulfalizer) [inline doc/guides/kconfig/index.rst]: Return/returns typo (though I had that in the original version too).

**Review by ulfalizer — APPROVED** [review]

File: doc/guides/kconfig/index.rst (hunk 6 above)
Comment (galak) [reply doc/guides/kconfig/index.rst]: fixed

**Review by Laczen — APPROVED** [review]
Nice solution, thanks

**Review by ioannisg — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

**Pull request merged**